File: CountPrefixAndSuffixPairsI.js
// https://leetcode.com/problems/count-prefix-and-suffix-pairs-i/

/**
 * @param {string[]} words
 * @return {number}
 */
var countPrefixSuffixPairs = function(words) {
    const isPrefixAndSuffix = (a, b) => {
        let x = 0;
        let y = b.length-a.length;
        for (let i = 0; i < a.length; i++){
            if (b[x] != a[i] || b[y] != a[i]){
                return false
            }
            x++
            y++
        }
        return true
    }
    let count = 0;
    for (let i = 0; i < words.length-1; i++){
        for (let j = i+1; j < words.length; j++ ){
            if (isPrefixAndSuffix(words[i], words[j])) count++
        }
    }
    return count
};

File: CountPrefixAndSuffixPairsI.cs
public class Solution {
    public int CountPrefixSuffixPairs(string[] words) {
        int count = 0;
        for (int i = 0; i < words.Length -1; i++)
        {
            for (int j = i + 1; j < words.Length; j++)
            {
                if (IsBothPrefixAndSuffix(words[i], words[j]))
                {
                    count++;
                }
            }
        }
        return count;
    }

    public bool IsBothPrefixAndSuffix(string fix, string word){
        if (fix.Length > word.Length) return false;
        for (int i = 0; i < fix.Length; i++)
        {
            if(fix[i] != word[i] || fix[i] != word[word.Length - fix.Length + i])
            {
                return false;
            }
        }
        return true;
    }

}